src/DelegateDecompiler.EntityFramework.Tests/TestGroup90AdditionalFeatures/Test01NestedExpressions.cs

```diff
@@ -68,5 +69,48 @@ public void TestSubqueryAsContextExtensionMethod()
                 env.CompareAndLogList(linq, dd);
             }
         }
+
+        [Test]
+#if EF_CORE && !EF_CORE3 && !EF_CORE5
+        [Ignore("Not natively supported in EF_CORE < 3")]
+#endif
+        public void TestSubqueryAsVariableReference()
+        {
+            using (var env = new MethodEnvironment(classEnv))
+            {
+                //ATTEMPT
+                env.AboutToUseDelegateDecompiler();
+
+                var referencedQuery = env.Db.Set<Animal>().Where(it => it.Species == "Canis lupus").Decompile();
```

Comment:
Can you add test where the query would not be decompiled

Reply:
Hi @hazzik, got a bit sidetracked.
I'm not sure I understand what test you want added here, so I added one failing test to expose the issue and refactored the other to compare materialized results